dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/schema/DubboBeanDefinitionParser.java

```diff
@@ -419,4 +403,32 @@ private static String resolveAttribute(Element element, String attributeName, Pa
         Environment environment = parserContext.getReaderContext().getEnvironment();
         return environment.resolvePlaceholders(attributeValue);
     }
+
+    private static String generateId(Element element, ParserContext parserContext, Class<?> beanClass, boolean required) {
+        String id = resolveAttribute(element, "id", parserContext);
+        if (StringUtils.isEmpty(id) && required) {
+            String generatedBeanName = resolveAttribute(element, "name", parserContext);
+
+            if (StringUtils.isEmpty((generatedBeanName))) {
+                if (ProtocolConfig.class.equals(beanClass)) {
+                    generatedBeanName = CommonConstants.DUBBO;
+                } else {
+                    generatedBeanName = resolveAttribute(element, CommonConstants.INTERFACE_KEY, parserContext);
+                    generatedBeanName = StringUtils.isEmpty(generatedBeanName) ? beanClass.getName() : generatedBeanName;
+                    if (beanClass.equals(ServiceBean.class)) {
+                        String groupName = resolveAttribute(element, CommonConstants.GROUP_KEY, parserContext);
```

Comment:
Here you need to check whether the group/version is empty, if it is empty, try to read from the provider config if the `provider` or `providerIds` attribute is not empty.  

NOTE: If present  `provider` attribure, it may be a  `RuntimeBeanReference` of nest config.
```
<dubbo:provider group="demo"  version="1.2.3" >
    <dubbo:service .../>
</dubbo:provider>
```

If present `providerIds`, it may be the provider bean name.

```
<dubbo:provider id="myprovider" group="demo"  version="1.2.3" />
<dubbo:service ... provider="myprovider" />
```

There are still some omissions, such as setting group/version in the default provider config. But it is more complicated to get the default provider config and may not be registered yet.